spirv_hlsl.cpp

```diff
@@ -1102,26 +1102,52 @@ void CompilerHLSL::emit_texture_op(const Instruction &i)
 		texop += "texelFetch";
 	else
 	{
-		texop += "tex2D";
-
-		if (gather)
-			texop += "Gather";
-		if (coffsets)
-			texop += "Offsets";
-		if (proj)
-			texop += "Proj";
-		if (grad_x || grad_y)
-			texop += "Grad";
-		if (lod)
-			texop += "Lod";
+		if (options.shader_model >= 40)
+		{
+			texop += to_expression(img);
+
+			if (gather)
+				texop += ".Gather";
+			else if (offset || coffsets || bias)
```

Comment:
coffsets is just constant offset vs. offset which is dynamic, for purposes of code-gen, they are the same.